packages/dd-trace/src/encode/agentless-ci-visibility.js

```diff
@@ -0,0 +1,312 @@
+'use strict'
+const tracerVersion = require('../../lib/version')
+const { truncateSpan, normalizeSpan } = require('./tags-processors')
+const Chunk = require('./chunk')
+
+const ENCODING_VERSION = 1
+
+function formatSpan (span) {
+  return {
+    type: span.type === 'test' ? 'test' : 'span',
+    version: ENCODING_VERSION,
+    content: normalizeSpan(truncateSpan(span))
+  }
+}
+
+class AgentlessCiVisibilityEncoder {
```

Comment:
this is pretty similar to the `0.4` encoder, which follows https://github.com/darrachequesne/notepack/blob/master/lib/encode.js, only with the added length considerations